### PR TITLE
feat: implement BMC password standard

### PR DESCRIPTION
**Pull request by stevekeay**
This replaces our "placeholder" get-bmc-creds workflow template with a working implementation.  It uses a single shared secret to generate BMC passwords that are unique on each server, but without having to store each individual password.

The passwords are generated based on a cryptographic hash function of the IP address and a static master key.

It relies on a bmc-master secret created elsewhere.

### CODE REVIEW

File: python/understack-workflows/understack_workflows/main/get_standard_bmc_creds.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+
+def base64(string):
+    return b64encode(string.encode()).decode()
+
+
+def require_environment_variables(*names):
+    missing_environment_variables = [name for name in names if os.getenv(name) is None]
+    if missing_environment_variables:
+        print(f"Please set envionment: {missing_environment_variables=}")
```
Comment (cardoe) [inline python/understack-workflows/understack_workflows/main/get_standard_bmc_creds.py:22]:
```suggestion
        print(f"Please set environment: {missing_environment_variables=}")
```

File: workflows/argo-events/workflowtemplates/get-bmc-creds.yaml [hunk 2]
```diff
@@ -19,43 +18,49 @@ spec:
...
+          - name: ip_address
       outputs:
         parameters:
           - name: secret
             valueFrom:
               path: "/tmp/output.txt"
-      script:
-        image: ghcr.io/rackerlabs/understack/argo-utils-python3.11.8:latest
+      container:
+        image: ghcr.io/rss-engineering/undercloud-deploy/argo-undercloud-rackspace3.11.8:latest
```
Comment (cardoe) [inline workflows/argo-events/workflowtemplates/get-bmc-creds.yaml:52]: This should be a container from this repo.

**Review by cardoe — COMMENTED** [review]
Overall this should be good. 2 comments made inline. The only comment I didn't make inline would have been to have `create_secret()` take in the `kube` client object to make testing with a mock possible.

File: workflows/argo-events/workflowtemplates/get-bmc-creds.yaml (hunk 2 above)
Comment (stevekeay) [reply workflows/argo-events/workflowtemplates/get-bmc-creds.yaml]: fixed in 5c8d61c

**Pull request merged**